test/unit/fetch-tool.js

```diff
@@ -27,3 +29,47 @@ test('send failure returns response.status', t => {
         t.equal(reason, 500);
     });
 });
+
+test('get success returns Uint8Array.body(response.arrayBuffer())', t => {
+    const text = 'successful response';
+    const encoding = 'utf-8';
+    const encoded = new TextEncoder().encode(text);
+    const decoder = new TextDecoder(encoding);
+
+    global.fetch = () => Promise.resolve({
+        ok: true,
+        arrayBuffer: () => Promise.resolve(encoded.buffer)
+    });
+
+    const tool = new FetchTool();
+    
+    return tool.get({url: 'url'}).then(result => {
+        t.equal(decoder.decode(result), text);
+    });
+});
+
+test('get with 404 response returns null data', t => {
+    global.fetch = () => Promise.resolve({
+        ok: false,
+        status: 404
+    });
+
+    const tool = new FetchTool();
+
+    return tool.get('url').then(result => {
+        t.equal(result, null);
+    });
+});
+
+test('get failure returns response.status', t => {
+    global.fetch = () => Promise.resolve({
+        ok: false,
+        status: 500
+    });
+
+    const tool = new FetchTool();
+
+    return tool.get({url: 'url'}).catch(reason => {
+        t.equal(reason, 500);
+    });
```

Comment:
I would suggest using `t.resolves` and `t.rejects` to test these promises... assuming that we're using a new enough `tap`. Right now I think this one would pass if the promise resolves (doesn't reject), since it won't get to the `t.equal` at all so there's no check to fail.

Reply:
Update: I checked the `tap` version and it looks like we're good.